DSPPlugCombinedVolcanoPlot/LabeledCombinedVolcanoPlot.R

```diff
@@ -0,0 +1,647 @@
+  yaxis$brk <- 10^-(yaxis$brk)
+  
+  # keep scientific notation if small enough pvalues when changing to character
+  yaxis$label <- format(yaxis$brk, trim=TRUE, digits=4,
+                        scientific=ifelse(maxPval < 0.0001, TRUE, FALSE), 
+                        drop0trailing=TRUE)
+  
+  # add threshold lines if thresholds are not NULL
+  if(!is.null(fc_thresh)){
+    ggFigure <- ggFigure + geom_vline(xintercept=fc_thresh, linetype="dotted")+
+      geom_vline(xintercept=-fc_thresh, linetype="dotted")+
+      annotate("text", x=fc_thresh+0.35, y=1,family=font_family, size=font_size/3,
+               label=paste0("FC=", round(2^fc_thresh, digits=2)))
+  }
+  if(!is.null(pval_thresh)){
+    ggFigure <- ggFigure + geom_hline(yintercept=pval_thresh, linetype="dotted")
+    yaxis <- rbind(yaxis, c(pval_thresh, paste0('pval=',pval_thresh)))
+  }
+  if(!is.null(fdr_thresh)){
+    ggFigure <- ggFigure + geom_hline(yintercept=fdr_pval, linetype="dotted")
+    yaxis <- rbind(yaxis, c(fdr_pval, paste0('FDR=',fdr_thresh)))
+  }
+  
+  # order yaxis in increasing value 
+  yaxis$brk <- as.numeric(yaxis$brk)
+  yaxis <- yaxis[order(yaxis$brk, decreasing=F),]
+  
+  # subset de to only contain genes to label on plot, either by user specified genes or top n_genes by pval
+  if(!is.null(gene_list)){
+    gene_labels <- subset(de, subset=Target.Name %in% gene_list)
+  }else{
+    # remove gene labels for genes below lowest pvalue cutoff
+    gene_labels <- de[which(de$Pvalue < min(fdr_pval, pval_thresh, na.rm = T)),]
+    
+    # only label genes above fc_thresh if set by user else only look at pvalue
+    if(!is.null(fc_thresh) & label_fc == FALSE){
+      gene_labels <- gene_labels[which(abs(gene_labels$Log2) > fc_thresh),]
+    }
+    
+    # only keep top # of genes by pvalue
+    if (n_genes_per_dataset == TRUE){
+      gene_labels <- do.call(rbind, lapply(split(gene_labels, gene_labels$dataset_name), function(x) head(x[order(x$Pvalue), ], n_genes)))
+        } else{
+      gene_labels <- gene_labels[head(order(gene_labels$Pvalue, decreasing=FALSE), n=n_genes),]  
+    }
+  }  
+  interactive=ifelse(output_format=="html", TRUE, FALSE)
+  if (interactive==FALSE){
+  # add gene labels to ggplot
+  ggFigure <- ggFigure + geom_text_repel(data=gene_labels, aes(x=Log2, y=Pvalue, label=Target.Name), 
+                                         family=font_family, force=5, fontface="bold", min.segment.length=0.1,
+                                         size=label_size)
+  }
+  # add y axis labels 
+  ggFigure <- ggFigure + scale_y_continuous(trans=change_axis_revlog_trans(base=10), breaks=as.numeric(yaxis$brk),
+                                            labels=yaxis$label)
+  
+  colnames(gene_labels) <- c("Target tag", "Target group memership/s", "Target Name", "Log2", 
+                             "Pvalue", "Adjusted pvalue", "-log10 pvalue", "-log10 adjusted pvalue",
+                             "FDR")
```

Comment:
@rvitancol This vector is the wrong size. You are missing 2 column names. 
It should probably be: `c("Target tag", "Target group memership/s", "Target Name", "Log2",  "Pvalue", "Adjusted pvalue", "-log10 pvalue", "-log10 adjusted value", "Source", "Dataset Name", "FDR")`

Reply:
@maddygriz I pushed the change, please check. Thank you!